src/beamlime/applications/handlers.py

```diff
@@ -143,23 +153,55 @@ def update_histogram(self, message: WorkflowResultUpdate) -> None:
         for name, data in content.items():
             self.plot_item(name, data)
 
+    def show(self):
+        """Show the figures in a grid layout."""
+        from plopp.widgets import Box, HBar, VBar
+
+        figures = list(self.figures.values())
+        n_rows = len(figures) // self.max_column + 1
+        return Box(
+            VBar(
+                [
+                    HBar(
+                        figures[
+                            i_row * self.max_column : i_row * self.max_column
+                            + self.max_column
+                        ]
+                    )
+                    for i_row in range(n_rows)
+                ]
+            )
+        )
```

Comment:
I think you don't need the `VBar` and `HBar`.
You should be able to use nested lists, e.g. `Box([[f1, f2, f3], [f4, f5]])` would give you 2 rows with 3 and 2 figs on each row, respectively.